webpack.config.js

```diff
@@ -0,0 +1,33 @@
+// Load the default @wordpress/scripts config object
+const defaultConfig = require( '@wordpress/scripts/config/webpack.config' );
+
+const legacyAdminFileNames = [
+	'google-product-category-fields',
+	'infobanner',
+	'metabox',
+	'modal',
+	'orders',
+	'product-categories',
+	'product-sets-admin',
+	'products-admin',
+	'settings-commerce',
+	'settings-sync',
+];
+
+const legacyAdminFileEntries = {};
+
+legacyAdminFileNames.forEach( ( name ) => {
+	legacyAdminFileEntries[ `admin/${ name }` ] = `./assets/js/admin/${ name }.js`;
+} );
+
+module.exports = {
+	...defaultConfig,
+	entry: {
+		'admin/index.js': './assets/js/admin/index.js',
```

Comment:
Do we need `admin/index.js`? I figure we can stick with the existing multi entry point approach. If we're planning on migrating to a different approach, e.g. one bundled `admin` script, that's fine, but I wouldn't assume that in this PR. Happy to be educated about best approach here :)

Is there a specific reason the existing entry points are "legacy"? I think this is reasonable, so I'm happy to stick with that approach (i.e. rename `legacyAdminFileEntries` => `adminFileEntries` etc).

Reply:
The multiple entry loaded by different `wp_enqueue_script()` calls approach doesn't seem that practical to me. WooCommerce is becoming a fully JS powered interface where separate pages are loaded by JS rather than a new PHP admin request. Meaning there's no opportunity to load separate scripts on different pages by PHP. 

I'll ask for some input from our JS experts in Slack.

"Legacy" as in jQuery dependant rather than React and vanilla JS.

Reply:
Sounds reasonable - I don't think this is a blocker at all. If you are proposing a change or improvement - would be good to get more buy in. At face value, moving to a single admin js and front-end js entry point seems like a good plan, but I'm curious to hear forward-looking best-practice recommendations from core, blocks, an wp-admin teams :)

Reply:
I agree it's not a blocker. Most likely we can re-visit when we starting doing some more JS related work for this plugin.

Reply:
I see the (different) `admin.js` approach is still there, so can you please document the proposed change, or stick with the multi entry point approach? As it stands, you're sneaking in a change in how JS assets are structured, without documenting it.

I see you got some input from the team - great - but since you're changing this, you need to document your design/intention here. Or if you're on the fence, build individual js files as before.

Reply:
Fair point, I'm happy to remove this and we can revisit this discussion when we need to add some React based JS to the plugin. I've made a few tweaks in 5fa94a1c420f13ad09f9dbb565428104786a2592.